cachito/workers/tasks/gitsubmodule.py

```diff
@@ -29,4 +29,4 @@ def add_git_submodules_as_package(request_id):
             "version": f"{sm.url}#{sm.hexsha}",
         }
         log.debug("Adding submodule '%s' as a package for Cachito request", sm.name)
-        update_request_with_package(request_id, package)
+        update_request_with_package(request_id, package, package_subpath=sm.path)
```

Comment:
In the later commit it is mentioned that _purls for git submodules point to a different repo, subpath is neither needed nor valid_ then what is the purpose of adding this here?

Reply:
IMO it's worth storing this information even if we don't currently need it. We may need subpaths for something in the future (and they *are* already included in the `.packages` part of a `/requests/{id}` response)

Reply:
Okay makes sense. I don't see any harm in storing this additional info anyhow

Reply:
Is this accurate though? Semantically, isn't the subpath of a package related to its own root directory?
It should be OK ot have a different semantic for submodules, but then we should document this somewhere.

Reply:
Well, since subpaths are kind of a new concept, I guess we get to decide the semantics now. For purls, the semantics are obvious, because you must be able to locate the package with only the purl. For the JSON representation of a request, it probably does not matter that much, the current semantics are "relative path to package from root of request repo."

WDYT?